src-cpp/rdkafkacpp_int.h

```diff
@@ -206,15 +328,29 @@ class MessageImpl : public Message {
           return rkmessage_;
   }
 
+  Headers* get_headers() {
+    return headers_;
+  }
+
   RdKafka::Topic *topic_;
   rd_kafka_message_t *rkmessage_;
   bool free_rkmessage_;
   /* For error signalling by the C++ layer the .._err_ message is
    * used as a place holder and rkmessage_ is set to point to it. */
   rd_kafka_message_t rkmessage_err_;
```

Comment:
Was this meant to be a pointer? Having the entire struct here for an error code seems wasteful.